Frameworks/UIKit/NSParagraphStyle.mm

```diff
@@ -38,42 +43,95 @@ + (NSWritingDirection)defaultWritingDirectionForLanguage:(NSString*)languageName
-    UNIMPLEMENTED();
-    return StubReturn();
+    if (self = [super init]) {
+        _alignment = [[decoder decodeObjectForKey:@"_NSParagraphStyle._alignment"] unsignedIntegerValue];
+        _firstLineHeadIndent = [decoder decodeFloatForKey:@"_NSParagraphStyle._firstLineHeadIndent"];
+        _headIndent = [decoder decodeFloatForKey:@"_NSParagraphStyle._headIndent"];
+        _tailIndent = [decoder decodeFloatForKey:@"_NSParagraphStyle._tailIndent"];
+        _lineBreakMode = [[decoder decodeObjectForKey:@"_NSParagraphStyle._lineBreakMode"] unsignedIntegerValue];
+        _maximumLineHeight = [decoder decodeFloatForKey:@"_NSParagraphStyle._maximumLineHeight"];
+        _minimumLineHeight = [decoder decodeFloatForKey:@"_NSParagraphStyle._minimumLineHeight"];
+        _lineSpacing = [decoder decodeFloatForKey:@"_NSParagraphStyle._lineSpacing"];
+        _paragraphSpacing = [decoder decodeFloatForKey:@"_NSParagraphStyle._paragraphSpacing"];
+        _paragraphSpacingBefore = [decoder decodeFloatForKey:@"_NSParagraphStyle._paragraphSpacingBefore"];
+        _baseWritingDirection =
+            (NSWritingDirection)[[decoder decodeObjectForKey:@"_NSParagraphStyle._baseWritingDirection"] unsignedIntegerValue];
+        _lineHeightMultiple = [decoder decodeFloatForKey:@"_NSParagraphStyle._lineHeightMultiple"];
+        _tabStops = [decoder decodeObjectForKey:@"_NSParagraphStyle._tabStops"];
+        _defaultTabInterval = [decoder decodeFloatForKey:@"_NSParagraphStyle._defaultTabInterval"];
+        _hyphenationFactor = [decoder decodeFloatForKey:@"_NSParagraphStyle._hyphenationFactor"];
+    }
+
+    return self;
 }
 
 /**
- @Status Stub
+ @Status Interoperable
 */
 - (void)encodeWithCoder:(NSCoder*)encoder {
-    UNIMPLEMENTED();
+    [encoder encodeObject:[NSNumber numberWithUnsignedInteger:_alignment] forKey:@"_NSParagraphStyle._alignment"];
+    [encoder encodeFloat:_firstLineHeadIndent forKey:@"_NSParagraphStyle._firstLineHeadIndent"];
+    [encoder encodeFloat:_headIndent forKey:@"_NSParagraphStyle._headIndent"];
+    [encoder encodeFloat:_tailIndent forKey:@"_NSParagraphStyle._tailIndent"];
+    [encoder encodeObject:[NSNumber numberWithUnsignedInteger:_lineBreakMode] forKey:@"_NSParagraphStyle._lineBreakMode"];
+    [encoder encodeFloat:_maximumLineHeight forKey:@"_NSParagraphStyle._maximumLineHeight"];
+    [encoder encodeFloat:_minimumLineHeight forKey:@"_NSParagraphStyle._minimumLineHeight"];
+    [encoder encodeFloat:_lineSpacing forKey:@"_NSParagraphStyle._lineSpacing"];
+    [encoder encodeFloat:_paragraphSpacing forKey:@"_NSParagraphStyle._paragraphSpacing"];
+    [encoder encodeFloat:_paragraphSpacingBefore forKey:@"_NSParagraphStyle._paragraphSpacingBefore"];
+    [encoder encodeObject:[NSNumber numberWithUnsignedInteger:_baseWritingDirection] forKey:@"_NSParagraphStyle._baseWritingDirection"];
+    [encoder encodeFloat:_lineHeightMultiple forKey:@"_NSParagraphStyle._lineHeightMultiple"];
```

Comment:
Consider this only a nit.

---
In reply to: [87270844](https://github.com/Microsoft/WinObjC/pull/1343#discussion_r87270844) [](ancestors = 87270844)